app/views/agents/show.html.erb

```diff
@@ -161,17 +161,19 @@
               <b>Options:</b>
               <pre><%= Utils.pretty_jsonify @agent.options || {} %></pre>
             </p>
-
-            <p id="memory" data-agent-id="<%= @agent.id %>">
-              <b>Memory:</b>
-              <% if @agent.memory.present? %>
-                <i class="fa fa-spinner fa-pulse spinner"></i>
-                <i class="fa fa-trash action-icon clear"></i>
-              <% end %>
-              <pre class="memory"><%= Utils.pretty_jsonify @agent.memory || {} %></pre>
-            </p>
+            <button type="submit" class="btn btn-default" id="target">Hide or Reveal Memory</button>
```

Comment:
with `if $('#memory').hasClass('hidden')` you could make it dependent from the shown text :smile: